package ICFDetector;

message LeafModel
{
	required int32 channel_index=1;
	required int32 x0=2;
	required int32 y0=3;
	required int32 x1=4;
	required int32 y1=5;
	required double threshold=6;
	required double score=7;/// get score if 'value'<threshold
}

message WeakModel
{
	required LeafModel l0_node=1;
	required LeafModel l1_node_small=2;
	required LeafModel l1_node_large=3;
}

message StrongModel
{
	repeated WeakModel weak_models=1;
	required int32 shrink_factor=2;
	required int32 model_width=3;
	optional int32 model_height=4 [default = -1];
}

message MultiScaleStrongModel
{
	required int32 shrink_factor = 1;
	required int32 scale0_model_width=2;	
	optional int32 scale0_model_height=5 [default = -1];
	repeated int32 models_width=3;
	repeated StrongModel strong_models=4;
}

